### PR TITLE
chore: consolidate redis code into core

### PR DESCRIPTION
**Pull request by buddy326**

### CODE REVIEW

**Comment by buddy326** [comment]
Im not done adding onto it, Im amending the commit as I go from one repo to another!

File: tests/test_redis.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
+from virtool_core.redis import connect
 import logging
 
 
 async def test_redis_success(caplog, redis_connection_string):
     caplog.set_level(logging.INFO)
-    await connect_to_redis(redis_connection_string)
+    await connect(redis_connection_string)
     for record in caplog.records:
+        print(record.message)
```
Comment (igboyes) [inline tests/test_redis.py:9]: Remove `print` statement.

File: virtool_core/redis.py [hunk 2]
```diff
@@ -53,25 +55,103 @@ async def connect_to_redis(
...
+    is not found in ``timeout`` seconds.
+
+    :param list_name: the name of the list to pop from
+    :param redis: the Redis client
+    :param timeout: seconds to wait before raising :class:``Timeout``
+    :return: the next job ID
+
+    """
+    logger.info(f"Waiting for a job for {timeout if timeout else 'infinity'} seconds")
+    return await asyncio.wait_for(get_next_job(list_name, redis), timeout)
```
Comment (igboyes) [inline virtool_core/redis.py:131]: This is specific to `virtool-core` and needs to stay there.

File: virtool_core/redis.py [hunk 3]
```diff
@@ -53,25 +55,103 @@ async def connect_to_redis(
...
+
+    :param list_name: the name of the list to pop from
+    :param redis: the Redis client
+    :return: the next job ID
+
+    """
+    result = await redis.blpop(list_name)
+
+    if result is not None:
+        return str(result[1], encoding="utf-8")
```
Comment (igboyes) [inline virtool_core/redis.py:146]: This is specific to `virtool-core` and needs to stay there.

File: virtool_core/redis.py [hunk 4]
```diff
@@ -53,25 +55,103 @@ async def connect_to_redis(
...
+
+async def wait_for_cancellation(redis, job_id: str, func: Callable):
+    (channel,) = await redis.subscribe(CANCELLATION_CHANNEL)
+
+    try:
+        async for cancelled_job_id in channel.iter():
+            if cancelled_job_id.decode() == job_id:
+                return func()
+    except CancelledError:
+        ...
```
Comment (igboyes) [inline virtool_core/redis.py:157]: This is specific to `virtool-core` and needs to stay there.

**Review by igboyes — CHANGES_REQUESTED** [review]
There is duplication between `connect` and `configure_redis`. Reuse `connect` in `configure_redis`.

File: tests/test_redis.py (hunk 1 above)
Comment (buddy326) [reply tests/test_redis.py:9]: I see where the duplication is happening but I am leaving configure_redis in virtool_core still though right as its not entirely the same as connect? and taking it out of virtool_workflow still?

**Review by igboyes — CHANGES_REQUESTED** [review]
* Use `redis_connection_string` only. Not `url`.
* Add `timeout` support to `configure_redis` and `connect`.
* Move logging to `connect` from `configure_redis` where it makes sense. Remove `Connected to Redis` message.

**Pull request merged**